pkg/chartmuseum/server/multitenant/api.go

```diff
@@ -109,6 +111,21 @@ func (server *MultiTenantServer) uploadChartPackage(log cm_logger.LoggingFn, rep
 			return &HTTPError{409, "file already exists"}
 		}
 	}
+
+	if server.EnforceSemver2 {
+		version, err := cm_repo.ChartVersionFromStorageObject(storage.Object{
+			Content: content,
+			// Since we only need content to check for the chart version
+			// left the others fields to be default
+		})
+		if err != nil {
+			return &HTTPError{400, err.Error()}
+		}
+		if _, err := semver.StrictNewVersion(version.Metadata.APIVersion); err != nil {
```

Comment:
`version.Metadata.APIVersion` would probably be "v1", and not what youre looking for. In fact, i think this will cause all uploads to fail. I think you need `version.Metadata.Version`